### ISSUE TITLE
cache pytorch_model.bin

### ISSUE THREAD

**Issue by johnml1135** [sillsdev/machine.py#25]
```
Downloading pytorch_model.bin:   0% 10.5M/2.46G [00:01<05:48, 7.04MB/s]
2023-08-23 20:30:27
Downloading pytorch_model.bin:   2% 52.4M/2.46G [00:07<05:30, 7.29MB/s]
2023-08-23 20:30:32
Downloading pytorch_model.bin:   3% 83.9M/2.46G [00:11<05:41, 6.96MB/s]
```

Really?  Why are we downloading this every time?  7MB/sec is not reading from a cache dir.

**Comment by johnml1135** [comment]
I think it's huggingface - and I don't know if the cache is setup correctly - https://huggingface.co/docs/datasets/cache.

**Comment by johnml1135** [comment]
```
    docker_internal_mounts {
        sdk_cache: "/clearml_agent_cache"
        apt_cache: "/var/cache/apt/archives"
        ssh_folder: "~/.ssh"
        ssh_ro_folder: "/.ssh"
        pip_cache: "/root/.cache/pip"
        poetry_cache: "/root/.cache/pypoetry"
        vcs_cache: "/root/.clearml/vcs-cache"
        venv_build: "~/.clearml/venvs-builds"
        pip_download: "/root/.clearml/pip-download-cache"
    }
```

We probably need `~/.cache/huggingface/datasets.`

**Comment by johnml1135** [comment]
More about huggingface cache: https://huggingface.co/docs/huggingface_hub/guides/manage-cache

**Comment by johnml1135** [comment]
This should be fixed on clearml - https://github.com/allegroai/clearml/issues/1101.

**Comment by mshannon-sil** [comment]
@johnml1135 I added the huggingface cache as an internal mount to the clearml.conf file on the AQuA server. Some files are now saving and loading from there, but pytorch_model.bin does not save there. Instead, it goes to `/root/machine/builds/[build_id]/model/pytorch_model.bin`. Should I cache `/root/machine/builds`, or would that interfere with any updates you would make to machine-jobs?

**Comment by mshannon-sil** [comment]
Looks like the `use_cache` parameter is set to the opposite of the `gradient_checkpointing` value in the machine config. The experiments on ClearML set `gradient_checkpointing` to True, so `use_cache` is being set to False. @ddaspit @johnml1135 do you know what the reason was for designing this behavior in the code? And is it necessary or can we just set `use_cache` to True?

**Comment by ddaspit** [comment]
Huggingface transformers [does not support](https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/models/m2m_100/modeling_m2m_100.py#L1034) `gradient_checkpointing=True` and `use_cache=True` at the same time. I would try temporarily disabling checkpointing and setting `use_cache` to `True` to see if that fixes the problem. Gradient checkpointing saves us some GPU memory. I'm not sure if we need it to train NLLB 1.3B or not.

**Comment by johnml1135** [comment]
We should probably use:
```
    extra_docker_arguments: ["--env","SIL_NLP_DATA_PATH=/aqua-ml-data",
        "--env","AWS_REGION=us-east-1",
        "--env","AWS_ACCESS_KEY_ID=xxxxxxxxxxxx",
        "--env","AWS_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxx"
        "-v","/home/clearml/.clearml/hf-cache:/root/.cache/huggingface" 
```

**Comment by mshannon-sil** [comment]
I added the extra docker argument to the server, and it's no longer downloading `pytorch_model.bin`.
